general-agent-module/src/function.calls.ts

```diff
@@ -0,0 +1,70 @@
+import { 
+  Finding, 
+  HandleTransaction, 
+  TransactionEvent, 
+  Trace,
+} from "forta-agent";
+import { FindingGenerator } from "./utils";
+import Web3 from "web3";
+
+const abi = new Web3().eth.abi;
+
+interface AgentOptions{
+  from?: string;
+  to?: string;
+  functionSignature: string;
+};
+
+interface TraceInfo{
+  from: string;
+  to: string;
+  input: string;
+};
+
+type Filter = (traceInfo: TraceInfo) => boolean,;
+
+const fromTraceActionToTraceInfo = (trace: Trace): TraceInfo => {
+  return {
+    to: trace.action.to,
+    from: trace.action.from,
+    input: trace.action.input,
+  };
+};
+
+const createFilter = (options: AgentOptions | undefined): Filter => {
+  if (options === undefined) {
+    return (_) => true;
+  }
+
+  return (traceInfo) => {
+    if (options.from !== undefined && options.from !== traceInfo.from) {
+      return false;
+    }
+
+    if (options.to !== undefined && options.to !== traceInfo.to) {
+      return false;
+    }
+
+    if (options.functionSignature !== undefined) {
+      const expectedSelector: string = abi.encodeFunctionSignature(options.functionSignature);
+      const functionSelector: string = traceInfo.input.slice(0, 10);
+      if(expectedSelector !== functionSelector)
+        return false;
+    }
+
+    return true;
+  };
+};
+
+export default function provideFunctionCallsDetectorAgent(
+  findingGenerator: FindingGenerator,
+  agentOptions?: AgentOptions
```

Comment:
`agentOptions` shouldn't be optional because the template needs the `functionSignature`. I would recommend moving the `functionSignature` from the options.